recipes/computer_vision/object_detection/model_server/object_detection_server.py

```diff
@@ -1,19 +1,30 @@
-from transformers import AutoImageProcessor, AutoModelForObjectDetection
+from transformers import AutoImageProcessor, AutoModelForObjectDetection, AutoConfig
+from huggingface_hub import snapshot_download
 from PIL import Image, ImageDraw
 from fastapi import FastAPI
 from pydantic import BaseModel
 import torch
 import base64
 import os
 import io
+import shutil
 
 
 app = FastAPI()
 model = os.getenv("MODEL_PATH", default="facebook/detr-resnet-101")
 revision = os.getenv("MODEL_REVISION", default="no_timm")
 
-processor = AutoImageProcessor.from_pretrained(model, revision=revision)
-model = AutoModelForObjectDetection.from_pretrained(model, revision=revision)
+if os.path.isfile(model):
```

Comment:
should this be an inverse condition? if model does not exist then download and copy

Reply:
No because AI Lab only provides a single file but several are needed so in this case we download from HF and then copy the AI Lab provided one